pkg/network/ebpf/c/runtime/tracer.c

```diff
@@ -398,15 +398,28 @@ int kprobe__skb_consume_udp(struct pt_regs *ctx) {
 SEC("kprobe/tcp_retransmit_skb")
 int kprobe__tcp_retransmit_skb(struct pt_regs *ctx) {
     struct sock *sk = (struct sock *)PT_REGS_PARM1(ctx);
-
-#if LINUX_VERSION_CODE < KERNEL_VERSION(4, 7, 0)
-    int segs = 1;
-#else
-    int segs = (int)PT_REGS_PARM3(ctx);
-#endif
     log_debug("kprobe/tcp_retransmit\n");
+    u64 tid = bpf_get_current_pid_tgid();
+    tcp_retransmit_skb_args_t args = {};
+    args.sk = sk;
+    args.segs = 0;
+    bpf_map_update_with_telemetry(pending_tcp_retransmit_skb, &tid, &args, BPF_ANY);
 
-    return handle_retransmit(sk, segs);
+    return 0;
+}
+
+SEC("kretprobe/tcp_retransmit_skb")
+int kretprobe__tcp_retransmit_skb(struct pt_regs *ctx) {
+    u64 tid = bpf_get_current_pid_tgid();
```

Comment:
you should handle the return code here, `if PT_REGS_RC(ctx) < 0 { bpf_map_delete_elem(&pending_tcp_retransmit_skb, &tid); return 0; }`

Reply:
thanks! will add